sdk/core/scheduler/main.cc

```diff
@@ -551,12 +595,19 @@ __cheriot_minimum_stack(0xa0) int futex_wake(uint32_t *address, uint32_t count)
 		  priority_boost_for_thread(currentThread->id_get()));
 		// If we have dropped priority below that of another runnable thread, we
 		// should yield now.
-		shouldYield |= !currentThread->is_highest_priority();
 	}
 
-	if (shouldYield)
+	switch (shouldYield)
 	{
-		yield();
+		case YieldLater:
+			Timer::ensure_tick();
+			break;
+		case YieldNow:
+			yield();
+			break;
+		case NoYield:
+		{
+		}
```

Comment:
Purely a stylistic question: why not `break;`?

Reply:
Also fine, I just wanted to make sure all cases were covered.  Break is probably better.